build/configuration-properties-collector-processor/src/main/java/org/hibernate/search/configuration/properties/collector/impl/AnnotationUtils.java

```diff
@@ -0,0 +1,64 @@
+/*
+ * Hibernate Search, full-text search for your domain model
+ *
+ * License: GNU Lesser General Public License (LGPL), version 2.1 or later
+ * See the lgpl.txt file in the root directory or <http://www.gnu.org/licenses/lgpl-2.1.html>.
+ */
+package org.hibernate.search.configuration.properties.collector.impl;
+
+import java.util.List;
+import java.util.Optional;
+import java.util.stream.Collectors;
+import javax.lang.model.element.AnnotationMirror;
+import javax.lang.model.element.AnnotationValue;
+import javax.lang.model.element.Element;
+
+import org.hibernate.search.util.common.impl.HibernateSearchConfiguration;
+
+
+public final class AnnotationUtils {
+
+	private AnnotationUtils() {
+	}
+
+	public static boolean isIgnored(Element element) {
+		return findAnnotation( element, HibernateSearchConfiguration.class )
+				.flatMap( a -> a.attribute( "ignore", boolean.class ) )
+				.orElse( false );
+	}
+
+	public static Optional<AnnotationAttributeHolder> findAnnotation(Element element, Class<?> annotation) {
+		for ( AnnotationMirror mirror : element.getAnnotationMirrors() ) {
+			if ( mirror.getAnnotationType().toString().equals( annotation.getName() ) ) {
+				return Optional.of( new AnnotationAttributeHolder( mirror ) );
+			}
+		}
+		return Optional.empty();
+	}
+
+	public static class AnnotationAttributeHolder {
+		private final AnnotationMirror annotationMirror;
+
+		private AnnotationAttributeHolder(AnnotationMirror annotationMirror) {
+			this.annotationMirror = annotationMirror;
+		}
+
+		@SuppressWarnings("unchecked")
+		public <T> Optional<T> attribute(String name, Class<T> klass) {
+			return annotationMirror.getElementValues().entrySet().stream()
+					.filter( entry -> entry.getKey().getSimpleName().contentEquals( name ) )
+					.map( entry -> ( (T) entry.getValue().getValue() ) )
+					.findAny();
+		}
+
+		@SuppressWarnings("unchecked")
+		public <T> Optional<List<T>> multiAttribute(String name, Class<T> klass) {
+			return annotationMirror.getElementValues().entrySet().stream()
+					.filter( entry -> entry.getKey().getSimpleName().contentEquals( name ) )
+					.map( entry -> ( (List<AnnotationValue>) entry.getValue().getValue() ) )
+					.map( list -> (List<T>) list.stream().map( AnnotationValue::getValue ).collect( Collectors.toList() ) )
```

Comment:
Same here, unchecked casts are bad.

```suggestion
		public <T> Optional<List<T>> multiAttribute(String name, Class<T> klass) {
			return annotationMirror.getElementValues().entrySet().stream()
					.filter( entry -> entry.getKey().getSimpleName().contentEquals( name ) )
					.map( entry -> ( (List<AnnotationValue>) entry.getValue().getValue() ) )
					.map( list -> list.stream().map( AnnotationValue::getValue ).map( klass::cast ).collect( Collectors.toList() ) )
```